src/components/buttons/default.tsx

```diff
@@ -0,0 +1,85 @@
+import * as React from 'react'
+import styled, { css } from 'styled-components'
+import colors from '../../assets/colors'
+import * as fonts from '../../assets/fonts'
+
+interface ButtonProps extends React.HTMLProps<Button> {
+    state?: ButtonState
+    block?: boolean
+}
+
+export class ButtonState extends String {
+    static Default: ButtonState = "default"
+    static Loading: ButtonState = "loading"
+    static Success: ButtonState = "success"
+    static Failure: ButtonState = "failure"
+}
+
+class Button extends React.Component<ButtonProps, any> {
+    public static defaultProps: ButtonProps
```

Comment:
I don't understand what this ends up being? Can you use a TS interface as an object and what keys/values does it contain?